lib/model/SignUpStatus.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:openfoodfacts/model/Status.dart';
+
+class SignUpStatus extends Status {
+  final SignUpStatusError? statusError;
+
+  SignUpStatus._({
+    dynamic status,
+    String? statusVerbose,
+    String? body,
+    String? error,
+    int? imageId,
+    this.statusError,
+  }) : super(
+          status: status,
+          statusVerbose: statusVerbose,
+          body: body,
+          error: error,
+          imageId: imageId,
+        );
+
+  // Sentences are available in
+  // [https://github.com/openfoodfacts/openfoodfacts-translations/blob/main/openfoodfacts-web/openfoodfacts-web-fr.po]
+  factory SignUpStatus(Status status) {
+    if (status.body == null) {
+      return SignUpStatus._(
+        status: 500,
+        statusError: SignUpStatusError.UNKNOWN,
+        error:
+            'No response, open an issue here: https://github.com/openfoodfacts/openfoodfacts-dart/issues/new',
+      );
+    } else if (status.body!.contains('loggedin')) {
+      return SignUpStatus._(
+        status: 201,
+      );
+    } else if (status.body!
```

Comment:
Instead of this long string of `if`s I would recommend something that starts like this:
```dart
final Map<String, SignUpStatusError> _errorTexts = <String, SignUpStatusError>{
  'This username already exists, please choose another.':
  	SignUpStatusError.USERNAME_ALREADY_USED,
  'The e-mail address is already used.':
  	SignUpStatusError.EMAIL_ALREADY_USED,
  'The password needs to be a least 6 characters long.':
  	SignUpStatusError.INVALID_PASSWORD
  'The password and confirmation password are different.':
  	SignUpStatusError.INVALID_PASSWORD,
  'The user name must contain only unaccented letters, digits and dashes.':
  	SignUpStatusError.INVALID_USERNAME,
};
```